File: CouponSystemGitHub/node.js/serverjs/beans/errorObject.js
/** 
 * a resource that contains the constructor for the general "Error" object in the system 
 * @author Yuval Maayan
 * @version 1.0
 * */

/**@param code = error status
  *@param message = the error's message 
  *@return an object to be sent when an error has occurd  */
exports.object = function(code, message){
	return {"code":code,"message":message};
};

File: CouponSystemGitHub/java/API-FrontEnd-Layer/Coupon_System_Project/src/com/sys/exception/handeling/ErrorObject.java
package com.sys.exception.handeling;

import javax.xml.bind.annotation.XmlRootElement;

/**an object that's sent whenever an exception is being intercepted
 * @author Yuval Maayan and Lior Lev
 * @version 1.0
 */

@XmlRootElement
public class ErrorObject {
/**@param errorMessage = the error's message, usually set with the intercepted exception's message 
 * @param errorCode = the error code that fits the exception cause*/
	private String errorMessage;
	private int errorCode;

/**an empty constructor */
	public ErrorObject(){}
	
/**a constructor filling all of the attributes
 * @param errorMessage = the error message
 * @param errorCode = the error code */
	public ErrorObject(String errorMessage, int errorCode) {
		this.errorMessage = errorMessage;
		this.errorCode = errorCode;
	}

	public String getErrorMessage() {
		return errorMessage;
	}

	public void setErrorMessage(String errorMessage) {
		this.errorMessage = errorMessage;
	}

	public int getErrorCode() {
		return errorCode;
	}

	public void setErrorCode(int errorCode) {
		this.errorCode = errorCode;
	}
	
	
}
